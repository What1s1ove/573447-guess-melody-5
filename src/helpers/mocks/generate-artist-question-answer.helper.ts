import { IArtistQuestionAnswer } from '~/common/interfaces/interfaces';
import { ArtistName } from '~/common/enums/enums';

const AVATAR_URL = `https://api.adorable.io/avatars/128`;

const generateArtistQuestionAnswer = (
  artist: ArtistName
): IArtistQuestionAnswer => ({
  artist,
  picture: `${AVATAR_URL}/${Math.random()}`,
});

export { generateArtistQuestionAnswer };
