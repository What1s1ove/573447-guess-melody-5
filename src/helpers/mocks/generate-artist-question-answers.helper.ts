import { generateArtistQuestionAnswer } from '~/helpers/mocks/generate-artist-question-answer.helper';
import { IArtistQuestionAnswer } from '~/common/interfaces/interfaces';

const generateArtistQuestionAnswers = (
  artists: string[]
): IArtistQuestionAnswer[] => {
  const artistQuestionAnswers = artists.map((artist) =>
    generateArtistQuestionAnswer(artist)
  );

  return artistQuestionAnswers;
};

export { generateArtistQuestionAnswers };
