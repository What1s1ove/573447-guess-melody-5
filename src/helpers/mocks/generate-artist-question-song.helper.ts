import { ArtistName } from '~/common/enums/enums';
import { IArtistQuestionSong } from '~/common/interfaces/game/game';

const generateArtistQuestionSong = (
  artist: ArtistName
): IArtistQuestionSong => ({
  artist,
  src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
});

export { generateArtistQuestionSong };
