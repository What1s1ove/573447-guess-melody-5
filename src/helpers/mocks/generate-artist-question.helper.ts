import { getRandomItem, getShuffledArray } from '~/helpers/array/array';
import { generateArtistQuestionSong } from '~/helpers/mocks/generate-artist-question-song.helper';
import { generateArtistQuestionAnswer } from '~/helpers/mocks/generate-artist-question-answer.helper';
import { generateArtistQuestionAnswers } from '~/helpers/mocks/generate-artist-question-answers.helper';
import { getUniqueAnswerTypes } from '~/helpers/mocks/get-unique-answer-types.helper';
import { IArtistQuestion } from '~/common/interfaces/interfaces';
import { ArtistName, QuestionType } from '~/common/enums/enums';

const CUT_GENRE_START_IDX = 0;
const ANSWERS_COUNT = 2;

const artists = Object.values(ArtistName);

const generateArtistQuestion = (): IArtistQuestion => {
  const artist = getRandomItem(artists);
  const song = generateArtistQuestionSong(artist);
  const answer = generateArtistQuestionAnswer(artist);
  const uniqueArtists = getUniqueAnswerTypes(artist, artists);
  const answers = generateArtistQuestionAnswers(
    getShuffledArray(uniqueArtists.slice(CUT_GENRE_START_IDX, ANSWERS_COUNT))
  );

  return {
    song,
    type: QuestionType.ARTIST,
    answers: [answer, ...answers],
  };
};

export { generateArtistQuestion };
