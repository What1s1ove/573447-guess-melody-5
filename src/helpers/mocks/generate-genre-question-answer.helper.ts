import { GenreType } from '~/common/enums/enums';
import { IGenreQuestionAnswer } from '~/common/interfaces/interfaces';

const generateGenreQuestionAnswer = (
  genre: GenreType
): IGenreQuestionAnswer => ({
  genre,
  src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
});

export { generateGenreQuestionAnswer };
