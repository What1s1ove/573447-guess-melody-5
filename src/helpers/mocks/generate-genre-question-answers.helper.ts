import { generateGenreQuestionAnswer } from '~/helpers/mocks/generate-genre-question-answer.helper';
import { GenreType } from '~/common/enums/enums';
import { IGenreQuestionAnswer } from '~/common/interfaces/interfaces';

const generateGenreQuestionAnswers = (
  genres: GenreType[]
): IGenreQuestionAnswer[] => {
  const genreQuestionAnswers = genres.map((genre) => {
    const genreQuestionAnswer = generateGenreQuestionAnswer(genre);

    return genreQuestionAnswer;
  });

  return genreQuestionAnswers;
};

export { generateGenreQuestionAnswers };
