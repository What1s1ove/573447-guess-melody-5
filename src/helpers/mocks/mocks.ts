export * from './generate-questions.helper';
export * from './generate-question.helper';
export * from './generate-genre-question-answer.helper';
export * from './generate-genre-question-answers.helper';
export * from './get-unique-genre-types.helper';
export * from './generate-genre-question.helper';
export * from './generate-artist-question-song.helper';
