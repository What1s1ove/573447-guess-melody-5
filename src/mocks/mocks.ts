export * from './game/games';
export * from './store/store';
