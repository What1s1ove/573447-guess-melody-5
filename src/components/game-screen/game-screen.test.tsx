import * as React from 'react';
import { render } from 'enzyme';
import { mockedQuestions } from '~/mocks/mocks';
import GameScreen from './game-screen';

describe(`GameScreen component`, () => {
  it(`should render GameScreen component`, () => {
    const component = render(<GameScreen questions={mockedQuestions} />);

    expect(component).toMatchSnapshot();
  });
});
