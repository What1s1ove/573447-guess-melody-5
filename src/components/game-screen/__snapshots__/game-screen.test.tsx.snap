// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameScreen component should render GameScreen component 1`] = `
<section
  class="game game--artist"
>
  <header
    class="game__header"
  >
    <a
      class="game__back"
      href="#"
    >
      <span
        class="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        class="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      class="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        class="timer__line"
        cx="390"
        cy="390"
        r="370"
        style="filter:url(#blur);transform:rotate(-90deg) scaleY(-1);transform-origin:center"
      />
    </svg>
    <div
      class="game__mistakes"
    >
      <div
        class="wrong"
      />
      <div
        class="wrong"
      />
      <div
        class="wrong"
      />
    </div>
  </header>
  <section
    class="game__screen"
  >
    <h2
      class="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      class="game__track"
    >
      <div
        class="track"
      >
        <button
          aria-label="Play track"
          class="track__button track__button--play"
          type="button"
        />
        <div
          class="track__status"
        >
          <audio
            src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
          />
        </div>
      </div>
    </div>
    <form
      class="game__artist"
    >
      <div
        class="artist"
      >
        <label
          class="artist__name"
          for="answer-0"
        >
          <input
            class="artist__input visually-hidden"
            id="answer-0"
            name="answer"
            type="radio"
          />
          <img
            alt="John Snow"
            class="artist__picture"
            src="https://api.adorable.io/avatars/128/0.1"
          />
          John Snow
        </label>
      </div>
      <div
        class="artist"
      >
        <label
          class="artist__name"
          for="answer-1"
        >
          <input
            class="artist__input visually-hidden"
            id="answer-1"
            name="answer"
            type="radio"
          />
          <img
            alt="Jim Beam"
            class="artist__picture"
            src="https://api.adorable.io/avatars/128/0.1"
          />
          Jim Beam
        </label>
      </div>
      <div
        class="artist"
      >
        <label
          class="artist__name"
          for="answer-2"
        >
          <input
            class="artist__input visually-hidden"
            id="answer-2"
            name="answer"
            type="radio"
          />
          <img
            alt="Jack Daniels"
            class="artist__picture"
            src="https://api.adorable.io/avatars/128/0.1"
          />
          Jack Daniels
        </label>
      </div>
    </form>
  </section>
</section>
`;
